boards/ikea-tradfri/board.c

```diff
@@ -20,6 +20,38 @@
 
 #include "board.h"
 #include "cpu.h"
+#include "mtd.h"
+#include "mtd_spi_nor.h"
+#include "timex.h"
+
+#ifdef MODULE_MTD
+static const mtd_spi_nor_params_t _ikea_tradfri_nor_params = {
+    .opcode = &mtd_spi_nor_opcode_default,
+    .wait_chip_erase = 2LU * US_PER_SEC,
+    .wait_32k_erase = 500LU *US_PER_MS,
+    .wait_sector_erase = 300LU * US_PER_MS,
+    .wait_4k_erase = 300LU * US_PER_MS,
+    .wait_chip_wake_up = 1LU * US_PER_MS,
+    .clk = IKEA_TRADFRI_NOR_SPI_CLK,
+    .flag = IKEA_TRADFRI_NOR_FLAGS,
+    .spi = IKEA_TRADFRI_NOR_SPI_DEV,
+    .mode = IKEA_TRADFRI_NOR_SPI_MODE,
+    .cs = IKEA_TRADFRI_NOR_SPI_CS,
+    .addr_width = 3,
+};
+
+static mtd_spi_nor_t ikea_tradfri_nor_dev = {
+    .base = {
+        .driver = &mtd_spi_nor_driver,
+        .page_size = IKEA_TRADFRI_NOR_PAGE_SIZE,
+        .pages_per_sector = IKEA_TRADFRI_NOR_PAGES_PER_SECTOR,
+        .sector_count = IKEA_TRADFRI_NOR_SECTOR_COUNT,
```

Comment:
If those may come with varying sized chips you could even leave that out now and it should be detected at run-time.

Reply:
So far, I haven't identified modules with other sizes.